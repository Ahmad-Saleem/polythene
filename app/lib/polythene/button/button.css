.button{
display:inline-block;
min-width:64px;
height:48px;
margin:0 4px;
padding:6px 0;
background:transparent;
text-align:center;
font:inherit;
text-transform:uppercase;
outline:none;
cursor:pointer;
-webkit-user-select:none;
-moz-user-select:none;
-ms-user-select:none;
user-select:none;
font-size:14px;
font-weight:500;}
.button .content{
position:relative;
height:36px;
line-height:36px;
padding:0 8px;
border-radius:2px;}
.button .content > span{
position:relative;
z-index:2;
pointer-events:auto;
-webkit-font-smoothing:antialiased;
-moz-osx-font-smoothing:grayscale;}
.button .ripple{
z-index:3;
top:-6px;
bottom:-6px;}
.button .ripple .ripple-mask{
top:6px;
bottom:6px;}
.button.selected,.button[disabled]{
cursor:default;
pointer-events:none;}
.button .wash{
z-index:1;
border-radius:inherit;
pointer-events:none;
transition:background-color 0.2s ease-in-out;}
