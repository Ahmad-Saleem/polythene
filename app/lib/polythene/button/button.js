define(["polythene/polythene/polythene","mithril","polythene/ripple/ripple","polythene/shadow/shadow","css!./button"],function(n,e,t,o){"use strict";var i,a,r,s,l;return i=function(i,a){var r,d,u,p,c;return a=a||{},a.ripple=a.ripple||{},a.shadow=a.shadow||{},r=a.tag||(a.url?"a":"div"),p=void 0!==a.ink&&!a.ink,a.disabled&&(r+="[disabled]"),p&&(r+="[noink]"),a.raised&&(r+="[raised]"),d={"class":[a.parentClass||"button",a["class"]].join(" ")},a.disabled||(d=Object.assign({},d,{config:function(n,e,t){e||(s(n,i,a.shadow),t.onunload=function(){l(n)})}},a.url?a.url:null)),c=null,a.label&&(c=e("span",a.label)),a.content&&(c=a.content),u=[e("div",{"class":"button-content"},[c,a.disabled||p?null:e.component(t,a.ripple),a.disabled||void 0!==a.wash&&!a.wash?null:e(".wash[fit]"),a.raised&&!a.disabled?e.component(o,{z:i.z(),animated:!0}):null])],e(r,d,n.insertContent(u,a))},s=function(n,t,o){var i=function(n){var i,a,r;if(i=5,a=t.baseZ(),r=o.increase||1,5!=a){var s=t.z();"down"===n?(s+=r,s=Math.min(s,i)):"up"==n&&(s-=r,s=Math.max(s,a)),t.z(s),e.redraw()}};a=function(){i("down")},r=function(){i("up")},n.addEventListener("mousedown",a),n.addEventListener("mouseup",r)},l=function(n){n.removeEventListener("mousedown",a),n.removeEventListener("mouseup",r)},{controller:function(n){return{view:e.prop(),baseZ:e.prop(n.z||1),z:e.prop(n.z||1)}},view:function(n,e){e=e||{};var t=n.view();return(!t||e.refresh)&&(t=i(n,e),n.view(t)),t}}});