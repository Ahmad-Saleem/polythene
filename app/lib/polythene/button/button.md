# Button

<a class="btn-demo" href="http://arthurclemens.github.io/Polythene-Examples/button.html">Demo</a>

Displays a text button with a [shadow](#shadow) and [ripple](#ripple) effect.

See also: [Icon Button](#icon-button) and [Floating Action Button](#fab)


## Usage

	var btn = require('polythene/button/button');

	var myBtn = m.component(btn, {
		label: 'Button',
		raised: true
	});

Add a URL:

	var myBtn = m.component(btn, {
		label: 'Button',
		raised: true,
		url: {href: 'index.html'}
	});


## Variations

* Buttons can be flat or raised. Using `raised` without specifying `z` gives the button a default shadow.
* The hover effect can be hidden with `wash: false`.
* The ripple effect on click can be hidden with `ink: false`.
* No icon in button, as this is not part of the Material Design guidelines; use [icon Button](#icon-button) instead


## Options

| **Parameter** |  **Mandatory** | **Type** | **Default** | **Description** |
| ------------- | -------------- | -------- | ----------- | --------------- |
| **tag** | optional | String | 'div' | HTML tag |
| **className** | optional | String |  | Extra CSS class appended to 'button' |
| **label** | required | String | | The button label |
| **url** | optional | Object | | Button URL; options object containing `href` and `config` |
| **wash** | optional | Boolean | true | Set to false to hide the effect on hover |
| **ink** | optional | Boolean | true | Set to false to disable the ripple effect on click/tap |
| **raised** | optional | Boolean | false | Shows a shadow; on button press the shadow depth is increased by 1 |
| **z** | optional | Number 0-5 | 1 | The shadow depth for a raised button; raised buttons have a default z of 1 |
| **disabled** | optional | Boolean | false | Disables the button |


## Default generated HTML

	<div raised="true" class="button">
		<div class="content">
			<span>Button label</span>
			<div fit="true" class="ripple constrained">
				<div class="ripple-mask">
					<div class="ripple-waves" style=""></div>
				</div>
			</div>
			<div fit="true" class="wash"></div>
			<div fit="true" class="shadow">
				<div fit="true" class="shadow-bottom shadow-bottom-z-1"></div>
				<div fit="true" class="shadow-top shadow-top-z-1"></div>
			</div>
		</div>
	</div>


## TODO

* Option to wait for ripple to finish

