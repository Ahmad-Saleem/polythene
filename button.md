# Button

Displays a text button with a [ripple](ripple.md) effect.

Button variations: [Raised Button](raisedButton.md), [Icon Button](icon-button.md), [Floating Action Button](fab.md)



## Usage

* [Usage with Mithril](mithril/button.md)
* [Usage with React](react/button.md)



## Options

### Common component options

| **Parameter** |  **Mandatory** | **Type** | **Default** | **Description** |
| ------------- | -------------- | -------- | ----------- | --------------- |
| **element**   | optional       | String   | "a"         | HTML element tag; may also be "button" |
| **className** | optional       | String   |             | Extra CSS class appended to `pe-text-button` |
| **style**     | optional       | Object   |             | For setting simple style attributes |
| **id**        | optional       | String   |             | HTML element id |
| **content**   | use `label` or `content` | String, hyperscript or component | | Alternative content; replaces children and ignores `label` |
| **before**    | optional       | String, hyperscript or component | | Extra content before main content; note that this content is placed left of subsequent elements with a lower stacking depth |
| **after**     | optional       | String, hyperscript or component | | Extra content after main content; note that this content is placed right of preceding elements with a higher stacking depth |
| **events**    | optional       | Object | | Options object containing one or more standard events such as `onclick` |
| **tabindex**  | optional       | Integer | 0 | Tab index |
| **tone**      | optional       | String: "dark" or "light" |  | Renders the component light on dark (sets class `pe-dark-tone`); use "light" to locally inverse (sets class `pe-light-tone`) |

### Button specific options

| **Parameter** |  **Mandatory** | **Type** | **Default** | **Description** |
| ------------- | -------------- | -------- | ----------- | --------------- |
| **label** | use `label` or `content` | String | | The button label |
| **url** | optional | Object with `href`, optionally `oncreate` (for Mithril) or `onClick` (for React) | | URL location; Mithril: for in-app route linking set `oncreate : m.route.link`; React: for in-app route linking use `onClick` and a router such as `react-router-dom` |
| **disabled** | optional | Boolean | false | Disables the button |
| **selected** | optional | Boolean | false | Set to true to show the button as selected |
| **formaction** | optional | String | | "The URI of a program that processes the information submitted by the button. If specified, it overrides the action attribute of the button"s form owner." [source:MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-formaction) |
| **inactive** | optional | Boolean | | Set to `true` to disable button events and ripple/wash effects |
| **inactivate** | optional | Number | | The number of seconds after tap/click when the button is inactive; useful to prevent double clicks |

### Button appearance options

| **Parameter** |  **Mandatory** | **Type** | **Default** | **Description** |
| ------------- | -------------- | -------- | ----------- | --------------- |
| **wash** | optional | Boolean | true | Set to false to hide the effect on hover |
| **ink** | optional | Boolean | true | Set to false to disable the ripple effect on click/tap |
| **ripple** | optional (valid if `ink` is `true`) | Options object | | Pass [ripple](../polythene-ripple) options to define ripple behavior |
| **increase** | optional | Number | 1 | The z-index increment/decrement on tap; note that the maximum z value is 5 |
| **borders** | optional | Boolean | false | Set to `true` to add a border; by default the border has no color - set border color to class `pe-button__content` to see the border |



### Composition

Button is composed from:

* [Ripple](ripple.md) (when option `ink` is not `false`)



## CSS classes

See: `polythene-core-button/src/classes.js`



## Future

* Option to wait before url/event is followed

