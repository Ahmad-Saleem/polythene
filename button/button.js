"use strict";function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{"default":obj}}Object.defineProperty(exports,"__esModule",{value:!0});var _typeof="function"==typeof Symbol&&"symbol"==typeof Symbol.iterator?function(obj){return typeof obj}:function(obj){return obj&&"function"==typeof Symbol&&obj.constructor===Symbol?"symbol":typeof obj};require("polythene/common/object.assign");var _polythene=require("polythene/polythene/polythene"),_polythene2=_interopRequireDefault(_polythene),_mithril=require("mithril"),_mithril2=_interopRequireDefault(_mithril),_ripple=require("polythene/ripple/ripple"),_ripple2=_interopRequireDefault(_ripple),_shadow=require("polythene/shadow/shadow"),_shadow2=_interopRequireDefault(_shadow);require("polythene/base-button/base-button"),require("polythene/button/theme/theme");var CSS_CLASSES={block:"pe-button pe-button--text",content:"pe-button__content",label:"pe-button__label",raised:"pe-button--raised",wash:"pe-button__wash",selected:"pe-button--selected",disabled:"pe-button--disabled",borders:"pe-button--borders",inactive:"pe-button--inactive"},MAX_Z=5,startType=window.PointerEvent?"pointerdown":"ontouchstart"in window||window.DocumentTouch&&document instanceof DocumentTouch?"touchstart":"mousedown",endType=window.PointerEvent?"pointerup":"ontouchend"in window||window.DocumentTouch&&document instanceof DocumentTouch?"touchend":"mouseup",tapStart=void 0,tapEnd=void 0,tapEndAll=void 0,downButtons=[],animateZ=function(ctrl,opts,name){var baseZ=ctrl.baseZ(),increase=opts.increase||1,z=ctrl.z();"down"===name&&5!==baseZ?(z+=increase,z=Math.min(z,MAX_Z)):"up"===name&&(z-=increase,z=Math.max(z,baseZ)),z!==ctrl.z()&&(ctrl.z(z),_mithril2["default"].redraw())},inactivate=function(ctrl,opts){ctrl.inactive=!0,_mithril2["default"].redraw(),setTimeout(function(){ctrl.inactive=!1,_mithril2["default"].redraw()},1e3*opts.inactivate)},initTapEvents=function(el,ctrl,opts){var tapHandler=function(ctrl,opts,name){"down"===name?downButtons.push({ctrl:ctrl,opts:opts}):"up"===name&&opts.inactivate&&!opts.inactive&&inactivate(ctrl,opts),opts.animateOnTap&&!_polythene2["default"].isTouch&&animateZ(ctrl,opts,name)};tapStart=function(){return tapHandler(ctrl,opts,"down")},tapEnd=function(){return tapHandler(ctrl,opts,"up")},tapEndAll=function(){downButtons.map(function(btn){tapHandler(btn.ctrl,btn.opts,"up")}),downButtons=[]},el.addEventListener(startType,tapStart),el.addEventListener(endType,tapEnd),window.addEventListener(endType,tapEndAll)},clearTapEvents=function(el){el.removeEventListener(startType,tapStart),el.removeEventListener(endType,tapEnd),window.removeEventListener(endType,tapEndAll)},createView=function(ctrl){var opts=arguments.length<=1||void 0===arguments[1]?{}:arguments[1],noink=void 0!==opts.ink&&!opts.ink,disabled=opts.disabled,inactive=ctrl.inactive,tag=opts.tag||"a",buttonConfig=function(el,isInited,context){isInited||disabled||inactive||(initTapEvents(el,ctrl,Object.assign({},opts,{animateOnTap:opts.animateOnTap!==!1})),context.onunload=function(){clearTapEvents(el)})},optsConfig=opts.config||function(){},urlConfig=opts.url&&opts.url.config?opts.url.config:function(){},props=Object.assign({},{"class":[opts.parentClass||CSS_CLASSES.block,opts.selected?CSS_CLASSES.selected:null,disabled?CSS_CLASSES.disabled:null,inactive?CSS_CLASSES.inactive:null,opts.borders?CSS_CLASSES.borders:null,opts.raised?CSS_CLASSES.raised:null,opts["class"]].join(" "),id:opts.id||""},opts.url?opts.url:null,opts.formaction?{formaction:opts.formaction}:null,opts.type?{type:opts.type}:null,opts.events?opts.events:null,{config:function(){return[buttonConfig.apply(void 0,arguments),optsConfig.apply(void 0,arguments),urlConfig.apply(void 0,arguments)]}},disabled?{disabled:!0}:null),label=opts.content?opts.content:opts.label?"object"===_typeof(opts.label)?opts.label:(0,_mithril2["default"])("div",{"class":CSS_CLASSES.label},opts.label):null,content=(0,_mithril2["default"])("div",{"class":CSS_CLASSES.content},[opts.raised&&!disabled?_mithril2["default"].component(_shadow2["default"],{z:ctrl.z(),animated:!0}):null,disabled||noink?null:_mithril2["default"].component(_ripple2["default"],opts.ripple||{}),disabled||void 0!==opts.wash&&!opts.wash?null:(0,_mithril2["default"])("div",{"class":CSS_CLASSES.wash}),label]);return(0,_mithril2["default"])(tag,props,[opts.before,content,opts.after])},component={controller:function(){var opts=arguments.length<=0||void 0===arguments[0]?{}:arguments[0],z=void 0!==opts.z?opts.z:1;return{baseZ:_mithril2["default"].prop(z),z:_mithril2["default"].prop(z),inactive:opts.inactive||!1}},view:function(ctrl){var opts=arguments.length<=1||void 0===arguments[1]?{}:arguments[1];return createView(ctrl,opts)}};exports["default"]=component,module.exports=exports["default"];
//# sourceMappingURL=button.js.map