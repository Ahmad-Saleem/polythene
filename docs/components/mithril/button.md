[Back to Polythene Button main page](../button.md)

# Button component for Mithril


## Options

[Button options](../button.md)


## Usage

~~~javascript
import m from "mithril"
import { Button } from "polythene-mithril"

m(Button, {
  label: "Button"
})
~~~

<a href="https://jsfiddle.net/ArthurClemens/5d5xfoxs/" target="_blank"><img src="http://arthurclemens.github.io/assets/polythene/docs/online-example.gif" height="36" /></a>


### Links

Add a route URL:

~~~javascript
m(Button, {
  label: "Button",
  url: {
    href: "/index",
    oncreate: m.route.link
  }
})
~~~

See also: [URLs and router links](../../handling-urls.md)


### Events

Add an onclick event:

~~~javascript
m(Button, {
  label: "Button",
  events: {
    onclick: () => console.log("click")
  }
})
~~~

### Variations

* The hover effect can be hidden with `wash: false`.
* The ripple effect on click can be hidden with `ink: false`.
* Button contains no icon as this is not part of the Material Design guidelines; use [Icon Button](../icon-button.md) instead.

### Mobile and tap delay

To remove the tap delay on mobile devices it is advisable to use a library like [Fastclick](https://github.com/ftlabs/fastclick). But because Fastclick has an unresolved issue with tap events while scrolling on iOS, it is better to use the convenience wrapper provided in "polythene-utilities". This temporarily removes the Fastclick event when an element is being scrolled.

~~~javascript
import { addFastClick } from "polythene-fastclick"

addFastClick()
~~~


## Appearance

### Styling

Below are examples how to change the Button appearance, either with a theme or with CSS.

You can find more information about theming in [Theming](../../theming.md).

#### Themed component

~~~javascript
Button.theme(".bordered-button", {
  color_light_text:   "#673ab7",
  color_light_border: "#673ab7",
  color_dark_text:    "yellow",
  color_dark_border:  "yellow"
})

m(Button, {
  label: "Borders",
  className: "bordered-button",
  borders: true
})
~~~

#### CSS

Change CSS using the CSS classes in `polythene-core-button/src/classes.js`

Class names can be imported with:

~~~javascript
import { classes } from "polythene-core-button";
~~~

#### Style

Some style attributes can be set using option `style`. For example:

~~~javascript
m(Button, {
  style: {
    backgroundColor: "#EF6C00",
    color:           "#fff"
  }
})
~~~

### Dark or light tone

If the component - or a component's parent - has option `tone` set to "dark", the component will be rendered with light colors on dark. 

* Use `tone: "dark"` to render light on dark
* Use `tone: "light"` to locally render normally when dark tone is set


