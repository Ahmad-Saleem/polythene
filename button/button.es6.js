import 'polythene/common/object.assign';
import p from 'polythene/polythene/polythene';
import m from 'mithril';
import ripple from 'polythene/ripple/ripple';
import shadow from 'polythene/shadow/shadow';
import 'polythene-theme/button/button';

let startType,
    endType,
    tapStart,
    tapEnd;

const initTapEvents = (el, ctrl, opts) => {
    const isTouch = p.isTouch();
    startType = isTouch ? 'click' : 'mousedown';
    endType = 'mouseup';
    // disable z animation on touch
    const animateOnTap = opts.animateOnTap && !isTouch;

    const tapHandler = function tapHandler(evt) {
        if (opts.onTap) {
            opts.onTap(evt);
        }
        if (!animateOnTap) {
            return;
        }
        const baseZ = ctrl.baseZ();
        if (baseZ === 5) {
            return;
        }
        const MAX_Z = 5;
        const increase = opts.increase || 1;
        let z = ctrl.z();
        if (evt === 'down') {
            z = z + increase;
            z = Math.min(z, MAX_Z);
        } else if (evt === 'up') {
            z = z - increase;
            z = Math.max(z, baseZ);
        }
        if (z !== ctrl.z()) {
            ctrl.z(z);
            m.redraw();
        }
    };
    tapStart = function(e) {
        e.preventDefault();
        tapHandler('down');
    };
    tapEnd = function(e) {
        e.preventDefault();
        tapHandler('up');
    };
    el.addEventListener(startType, tapStart);
    document.body.addEventListener(endType, tapEnd);
};

const clearTapEvents = function(el) {
    el.removeEventListener(startType, tapStart);
    document.body.removeEventListener(endType, tapEnd);
};

const createView = (ctrl, opts = {}) => {
    let tag, label;
    opts.ripple = opts.ripple || {};
    opts.shadow = opts.shadow || {};
    opts.animateOnTap = (opts.animateOnTap !== undefined) ? opts.animateOnTap : true;

    tag = opts.tag || 'a';
    const noink = opts.ink !== undefined && !opts.ink;
    const disabled = !!opts.disabled;

    if (disabled) {
        tag += '[disabled]';
    }
    if (noink) {
        tag += '.noink';
    }
    if (opts.raised) {
        tag += '.raised';
    }
    const props0 = {
        'class': [opts.parentClass || 'button', opts.selected ? 'selected' : null, opts.disabled ? 'disabled' : null, opts.class].join(' '),
        id: opts.id || ''
    };

    // handle multiple configs:
    // - passed as param in the url Object
    // - passed as opts.config
    // - the default buton config
    const buttonConfig = (el, isInited, context) => {
        if (isInited) {
            return;
        }
        initTapEvents(el, ctrl, opts);
        context.onunload = function() {
            clearTapEvents(el);
        };
    };
    const optsConfig = opts.config || (() => {});
    const urlConfig = (opts.url && opts.url.config) ? opts.url.config : (() => {});

    const props = Object.assign({},
        props0,
        opts.url ? opts.url : null,
        opts.events ? opts.events : null, {
            config: (...args) => [
                buttonConfig(...args),
                optsConfig(...args),
                urlConfig(...args)
            ]
        }
    );
    label = null;
    if (opts.content) {
        label = opts.content;
    } else if (opts.label) {
        if (typeof opts.label === 'object') {
            label = opts.label;
        } else {
            label = m('.label', opts.label);
        }
    }
    const content = m('div', {
        'class': 'content'
    }, [
        label,
        disabled || noink ? null : m.component(ripple, opts.ripple),
        disabled || opts.wash !== undefined && !opts.wash ? null : m('.wash.fit'),
        opts.raised && !disabled ? m.component(shadow, {
            z: ctrl.z(),
            animated: true
        }) : null
    ]);

    return m(tag, props, p.insertContent(content, opts));
};

const component = {
    controller: (opts = {}) => {
        let z = (opts.z !== undefined) ? opts.z : 1;

        return {
            baseZ: m.prop(z),
            z: m.prop(z)
        };
    },
    view: (ctrl, opts = {}) => {
        return createView(ctrl, opts);
    }
};

export default component;
