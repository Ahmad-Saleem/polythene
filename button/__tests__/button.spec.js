import m from 'mithril';
import {matchSnapshot} from '../../scripts/testing';
import button from 'polythene/button/button';

describe('Button component', () => {
    it('should render with no params', () => {
        const cmp = m(button);
        matchSnapshot(cmp);
    });
    it('should render params before and after', () => {
        const cmp = m(button, {
            before: m('div', 'BEFORE'),
            after: m('div', 'AFTER')
        });
        matchSnapshot(cmp);
    });
    it('should render common component params', () => {
        const cmp = m(button, {
            id: 'ID',
            tag: 'button',
            class: 'site--button'
        });
        matchSnapshot(cmp);
    });
    it('should render main params', () => {
        const cmp = m(button, {
            label: 'Send',
            borders: true,
            disabled: true,
            selected: true,
            formaction: '/submit',
            animateOnTap: true,
            inactive: true
        });
        matchSnapshot(cmp);
    });
    it('should render button appearance params (on)', () => {
        const cmp = m(button, {
            wash: true,
            ink: true,
            ripple: true,
            raised: true,
            z: 1
        });
        matchSnapshot(cmp);
    });
    it('should render button appearance params (off)', () => {
        const cmp = m(button, {
            wash: false,
            ink: false,
            ripple: false,
            raised: false
        });
        matchSnapshot(cmp);
    });
    it('should render param tabindex', () => {
        const cmp = m(button, {
            tabindex: 1
        });
        matchSnapshot(cmp);
    });
    it('should render param url', () => {
        const cmp = m(button, {
            url: {href: '/'}
        });
        matchSnapshot(cmp);
    });
    it('should render param content', () => {
        const cmp = m(button, {
            content: m('div', 'Send')
        });
        matchSnapshot(cmp);
    });

});