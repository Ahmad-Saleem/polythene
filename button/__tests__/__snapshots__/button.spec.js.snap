exports[`Button component should render button appearance params (off) 1`] = `
"<a class=\"pe-button pe-button--text       \" id=\"\" tabindex=\"0\">
  <div class=\"pe-button__content\">
    <div class=\"pe-ripple pe-ripple--constrained \" id=\"\">
      <div class=\"pe-ripple__mask\">
        <div class=\"pe-ripple__waves\">
        </div>
      </div>
    </div>
    <div class=\"pe-button__focus\">
    </div>
  </div>
</a>"
`;

exports[`Button component should render button appearance params (on) 1`] = `
"<a class=\"pe-button pe-button--text     pe-button--raised  \" id=\"\" tabindex=\"0\">
  <div class=\"pe-button__content\">
    <div class=\"pe-shadow pe-shadow--animated \" id=\"\">
      <div class=\"pe-shadow__bottom pe-shadow--z-1\">
      </div>
      <div class=\"pe-shadow__top pe-shadow--z-1\">
      </div>
    </div>
    <div class=\"pe-ripple pe-ripple--constrained \" id=\"\">
      <div class=\"pe-ripple__mask\">
        <div class=\"pe-ripple__waves\">
        </div>
      </div>
    </div>
    <div class=\"pe-button__wash\">
    </div>
    <div class=\"pe-button__focus\">
    </div>
  </div>
</a>"
`;

exports[`Button component should render common component params 1`] = `
"<button class=\"pe-button pe-button--text       site--button\" id=\"ID\" tabindex=\"0\">
  <div class=\"pe-button__content\">
    <div class=\"pe-ripple pe-ripple--constrained \" id=\"\">
      <div class=\"pe-ripple__mask\">
        <div class=\"pe-ripple__waves\">
        </div>
      </div>
    </div>
    <div class=\"pe-button__wash\">
    </div>
    <div class=\"pe-button__focus\">
    </div>
  </div>
</button>"
`;

exports[`Button component should render main params 1`] = `
"<a class=\"pe-button pe-button--text pe-button--selected pe-button--disabled pe-button--inactive pe-button--borders   \" id=\"\" tabindex=\"-1\" formaction=\"/submit\" disabled>
  <div class=\"pe-button__content\">
    <div class=\"pe-button__label\">
      Send
    </div>
  </div>
</a>"
`;

exports[`Button component should render param content 1`] = `
"<a class=\"pe-button pe-button--text       \" id=\"\" tabindex=\"0\">
  <div class=\"pe-button__content\">
    <div class=\"pe-ripple pe-ripple--constrained \" id=\"\">
      <div class=\"pe-ripple__mask\">
        <div class=\"pe-ripple__waves\">
        </div>
      </div>
    </div>
    <div class=\"pe-button__wash\">
    </div>
    <div class=\"pe-button__focus\">
    </div>
    <div>
      Send
    </div>
  </div>
</a>"
`;

exports[`Button component should render param tabindex 1`] = `
"<a class=\"pe-button pe-button--text       \" id=\"\" tabindex=\"1\">
  <div class=\"pe-button__content\">
    <div class=\"pe-ripple pe-ripple--constrained \" id=\"\">
      <div class=\"pe-ripple__mask\">
        <div class=\"pe-ripple__waves\">
        </div>
      </div>
    </div>
    <div class=\"pe-button__wash\">
    </div>
    <div class=\"pe-button__focus\">
    </div>
  </div>
</a>"
`;

exports[`Button component should render param url 1`] = `
"<a class=\"pe-button pe-button--text       \" id=\"\" tabindex=\"0\" href=\"/\">
  <div class=\"pe-button__content\">
    <div class=\"pe-ripple pe-ripple--constrained \" id=\"\">
      <div class=\"pe-ripple__mask\">
        <div class=\"pe-ripple__waves\">
        </div>
      </div>
    </div>
    <div class=\"pe-button__wash\">
    </div>
    <div class=\"pe-button__focus\">
    </div>
  </div>
</a>"
`;

exports[`Button component should render params before and after 1`] = `
"<a class=\"pe-button pe-button--text       \" id=\"\" tabindex=\"0\">
  <div>
    BEFORE
  </div>
  <div class=\"pe-button__content\">
    <div class=\"pe-ripple pe-ripple--constrained \" id=\"\">
      <div class=\"pe-ripple__mask\">
        <div class=\"pe-ripple__waves\">
        </div>
      </div>
    </div>
    <div class=\"pe-button__wash\">
    </div>
    <div class=\"pe-button__focus\">
    </div>
  </div>
  <div>
    AFTER
  </div>
</a>"
`;

exports[`Button component should render with no params 1`] = `
"<a class=\"pe-button pe-button--text       \" id=\"\" tabindex=\"0\">
  <div class=\"pe-button__content\">
    <div class=\"pe-ripple pe-ripple--constrained \" id=\"\">
      <div class=\"pe-ripple__mask\">
        <div class=\"pe-ripple__waves\">
        </div>
      </div>
    </div>
    <div class=\"pe-button__wash\">
    </div>
    <div class=\"pe-button__focus\">
    </div>
  </div>
</a>"
`;
